### PR TITLE
[PR #1537 Patch 2] Make MXNet installation optional for TextPredictor and AutoMMPredictor

### PR DESCRIPTION
**Pull request by zhiqiangdon**
*Previously, MXNet must be installed to use TextPredictor. This PR aims to make its installation optional.*

- Renamed `presets.py` -> `legacy_presets.py` (used by MXNet-based TextPredictor)
- Renamed `text_presets.py` - > `presets.py` (used by Pytorch-based TextPredictor)
- Fixed several importing issues after renaming files.
- Made unit tests for MXNet-based TextPredictor optional.
- Suppressed one huggingface warning "Some weights of the model checkpoint at google/electra-base-discriminator were not used when initializing ElectraModel: ['discriminator_predictions.dense_prediction.weight', 'discriminator_predictions.dense.bias', 'discriminator_predictions.dense_prediction.bias', 'discriminator_predictions.dense.weight']. This IS expected if you are initializing ElectraModel from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model). This IS NOT expected if you are initializing ElectraModel from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model)."

### CODE REVIEW

File: text/src/autogluon/text/text_prediction/__init__.py [hunk 1]
```diff
@@ -3,11 +3,10 @@
...
-from . import text_presets
 from .predictor import TextPredictor
 
 __all__ = [
     'constants', 'infer_types', 'metrics',
-    'predictor', 'presets', 'text_presets',
+    'predictor', 'presets',
     'TextPredictor',
-]
+]
```
Comment (sxjscience) [inline text/src/autogluon/text/text_prediction/__init__.py:12]: New line.

**Comment by sxjscience** [comment]
@zhiqiangdon You may also pick the fix in https://github.com/awslabs/autogluon/pull/1556

File: text/src/autogluon/text/text_prediction/__init__.py (hunk 1 above)
Comment (zhiqiangdon) [reply text/src/autogluon/text/text_prediction/__init__.py]: Added.

**Comment by szha** [comment]
Job PR-1557-6 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1557/6/index.html

**Review by sxjscience — APPROVED** [review]

**Pull request merged**